src/createDispatcher.js

```diff
@@ -50,19 +56,27 @@ export default function createDispatcher(opts = {}) {
 
   const logging = parseOpts(opts.logging)
   if (logging.agendas) {
+    // Seems like this means that middlewares are effectively an attachment on dispatchers
     logAgendas(dispatcher)
   }
 
   function next(agenda) {
     const obs = agenda
       .subscribeOn(scheduler)
-      .publishReplay()
+      .publishReplay() // Would this make all my actions replayable?
+                       // What if they are expensive or can't be performed again?
+                       // I definitely need to look up publishReplay when I have internet again
       .refCount()
 
     dispatcher.next(obs)
   }
 
   function dispatch(action) {
+    // Really enjoying the forced assertion pattern here, it's what we use in zmq.
+    // Fail fast and hard at the level of library
+    //
+    // People will inevitably have errors that crop up in userland. How should
+    // they handle these thrown errors for display to user?
```

Comment:
I think for this error it's not an issue to throw. Generally I try to only throw when the user violates an invariant or does something that is an anti pattern. We need to make sure though, that it is never necessary to try catch